map.go

```diff
@@ -717,13 +754,34 @@ func (mi *MapIterator) Next(keyOut, valueOut interface{}) bool {
 		return false
 	}
 
+	// gessing a random start key
+	if !supportNullKey.value && mi.randomStartGuess {
+		mi.randomStartGuess = false
+
+		var bytes1, bytes2, bytes3 []byte
+		bytes1, mi.err = mi.target.NextKeyBytes(mi.target.randKey)
+		if mi.err != nil {
+			return false
+		}
+		mi.prevKey = mi.target.randomizeKey()
+		bytes2, mi.err = mi.target.NextKeyBytes(mi.prevKey)
+		if !bytes.Equal(bytes1, bytes2) {
+			mi.prevKey = mi.target.randomizeKey()
+			bytes3, mi.err = mi.target.NextKeyBytes(mi.prevKey)
```

Comment:
Why do we need to check a third key, if key1 and key2 are not the same?

Reply:
if 1st random and 2nd random key lookup return the same value
   then we know both random key lookup to the 1st element of the map
Now if one (1st or 2nd) hit a key in the map, let's try a 3rd lookup 
   if 3rd match 1st or 2nd then we know that 3rd hit the 1st element of the map
   else we have no luck (that mean 2 random keys point to 2 elements on the map) and can't find a good way to start to iterate on the map (shouldn't happen (2 hits over 2^32) but technically exist)